skopt/acquisition.py

```diff
@@ -0,0 +1,203 @@
+
+    * `y_opt` [float, optional]:
+        The previous best value over which we want to improve.
+        Useful only when ``method`` is set to "EI"
+
+    * `method` [string, "LCB", "EI" or "PI", default="LCB"]:
+        "LCB": Lower Confidence Bound.
+        "EI": Expected Improvement.
+        "PI": Probability of Improvement.
+
+    * `xi`: [float, default=0.01]:
+        Controls how much improvement one wants over the previous best
+        values. Useful only when ``method`` is set to "EI"
+
+    * `kappa`: [float, default 1.96]:
+        Controls how much of the variance in the predicted values should be
+        taken into account. If set to be very high, then we are favouring
+        exploration over exploitation and vice versa.
+        Useless if ``method`` is set to "LCB".
+
+    Returns
+    -------
+    * `values`: [array-like, shape=(X.shape[0],)]:
+        Acquisition function values computed at X.
+    """
+    # Check inputs
+    X = np.asarray(X)
+    if X.ndim != 2:
+        raise ValueError("X should be 2-dimensional.")
+
+    # Evaluate acquisition function
+    if method == "LCB":
+        return gaussian_lcb(X, model, kappa)
+    elif method == "EI":
+        return gaussian_ei(X, model, y_opt, xi)
+    elif method == "PI":
+        return gaussian_pi(X, model, y_opt, xi)
+    else:
+        raise ValueError("Acquisition function not implemented.")
+
+    return values
+
+
+def gaussian_lcb(X, model, kappa=1.96):
+    """
+    Use the lower confidence bound to estimate the acquisition
+    values.
+
+    The trade-off between exploitation and exploration is left to
+    be controlled by the user through the parameter ``kappa``.
+
+    Parameters
+    ----------
+    * `X` [array-like, shape=(n_samples, n_features)]:
+        Values where the acquisition function should be computed.
+
+    * `model` [sklearn estimator that implements predict with ``return_std``]:
+        The fit estimator that approximates the function through the
+        method ``predict``.
+        It should have a ``return_std`` parameter that returns the standard
+        deviation.
+
+    * `kappa`: [float, default 1.96]:
+        Controls how much of the variance in the predicted values should be
+        taken into account. If set to be very high, then we are favouring
+        exploration over exploitation and vice versa.
+        Useless if ``method`` is set to "LCB".
+
+    Returns
+    -------
+    * `values`: [array-like, shape=(X.shape[0],)]:
+        Acquisition function values computed at X.
+    """
+    # Compute posterior.
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        mu, std = model.predict(X, return_std=True)
+
+    return mu - kappa * std
+
+
+def gaussian_pi(X, model, y_opt=0, xi=0.01):
+    """
+    Use the probability of improvement to calculate the acquisition values.
+
+    The predictions form a gaussian with a certain mean and standard deviation
+    as given by the model.
```

Comment:
I would rather say something like "The conditional probability p(y=f(x)|x) is assumed to be gaussian distributed, with mean and standard deviation approximated by `model`."